artifactory/spec/specfiles.go

```diff
@@ -137,50 +140,57 @@ func ValidateSpec(files []File, isTargetMandatory, isSearchBasedSpec bool) error
 		if !isSearchBasedSpec && !isPattern {
 			return errors.New("Spec must include a pattern.")
 		}
-		if isSearchBasedSpec && !isAql && !isPattern && !isBuild {
-			return errors.New("Spec must include either aql, pattern or build.")
+		if isBuild && isBundle {
+			return fileSpecCannotIncludeBothPropertiesValidation("build", "bundle")
+		}
+		if isSearchBasedSpec {
+			if !isAql && !isPattern && !isBuild && !isBundle {
+				return errors.New("Spec must include either aql, pattern or build.")
+			}
+			if isOffset {
+				if isBuild {
+					return fileSpecCannotIncludeBothPropertiesValidation("build", "offset")
+				}
+				if isBundle {
+					return fileSpecCannotIncludeBothPropertiesValidation("bundle", "offset")
+				}
+			}
+			if isLimit {
+				if isBuild {
+					return fileSpecCannotIncludeBothPropertiesValidation("build", "limit")
+				}
+				if isBundle {
+					return fileSpecCannotIncludeBothPropertiesValidation("bundle", "limit")
+				}
+			}
 		}
 		if isAql && isPattern {
-			return errors.New(fmt.Sprintf(fileSpecCannotIncludeBothPropertiesValidationMessage, "aql", "pattern"))
+			return fileSpecCannotIncludeBothPropertiesValidation("aql", "pattern")
 		}
 		if isAql && isExcludePatterns {
-			return errors.New(fmt.Sprintf(fileSpecCannotIncludeBothPropertiesValidationMessage, "aql", "exclude-patterns"))
+			return fileSpecCannotIncludeBothPropertiesValidation("aql", "exclude-patterns")
 		}
 		if isAql && isExclusions {
-			return errors.New(fmt.Sprintf(fileSpecCannotIncludeBothPropertiesValidationMessage, "aql", "exclusions"))
+			return fileSpecCannotIncludeBothPropertiesValidation("aql", "exclusions")
 		}
 		if isExclusions && isExcludePatterns {
-			return errors.New(fmt.Sprintf(fileSpecCannotIncludeBothPropertiesValidationMessage, "exclusions", "exclude-patterns"))
+			return fileSpecCannotIncludeBothPropertiesValidation("exclusions", "exclude-patterns")
 		}
 		if !isSortBy && isSortOrder {
 			return errors.New("Spec cannot include 'sort-order' if 'sort-by' is not included")
 		}
 		if isSortOrder && !isValidSortOrder {
 			return errors.New("The value of 'sort-order' can only be 'asc' or 'desc'.")
 		}
-		if isBuild && isSearchBasedSpec {
-			if err := validateFileSpecWithBuild(file); err != nil {
-				return err
-			}
-		}
 	}
 	if excludePatternsUsed {
 		showDeprecationOnExcludePatterns()
 	}
 	return nil
 }
 
-func validateFileSpecWithBuild(file File) error {
-	isOffset := file.Offset > 0
-	isLimit := file.Limit > 0
-
-	if isOffset {
-		return errors.New(fmt.Sprintf(fileSpecCannotIncludeBothPropertiesValidationMessage, "build", "offset"))
-	}
-	if isLimit {
-		return errors.New(fmt.Sprintf(fileSpecCannotIncludeBothPropertiesValidationMessage, "build", "limit"))
-	}
-	return nil
+func fileSpecCannotIncludeBothPropertiesValidation(fieldA, fieldB string) error {
```

Comment:
Since this method does not do validation, and also is not specific for properties, I suggest renaming it to **fileSpecValidationError**.